sway-lib-core/src/raw_slice.sw

```diff
@@ -0,0 +1,34 @@
+library raw_slice;
+
+dep raw_ptr;
+
+use ::raw_ptr::*;
+
+pub trait AsSlice {
+    fn as_slice(self) -> raw_slice;
+}
```

Comment:
What would the use for `raw_slice` be when we have `&[T]`? I thought it would just replace `raw_slice` so I even removed the space I was leaving for it :D

Reply:
IIUC, we don't want `raw_ptr` or `raw_slice` in Sway because they are too unsafe/low level, but we are just implementing them to unblock certain things until we can get an RFC on something better. I personally haven't worked on any Sway code that you'd want these instead of their typed variants.

Reply:
TBH, I don't even want this documented, but naming it `raw` would make that less likely so I've made the changes :D